### DISCUSSION TITLE
Report time dimension creation issues

### DISCUSSION THREAD

**Discussion by 936447225** [amzn/ads-advanced-tools-docs#146] [Q&A]
Is there any way to obtain hourly level reports by creating v3 reports for SP or v2 reports for SB or SD types? I think the backend supports this

**Comment by elizafarley** [comment]
Hi @936447225, the reporting APIs only support daily grain data at this time. If you want hourly data, I'd suggest looking into [Amazon Marketing Stream](https://advertising.amazon.com/API/docs/en-us/guides/amazon-marketing-stream/overview), which delivers data for SP, SB, and SD to your AWS account at an hourly grain.

**Reply by 936447225** [reply]
您好，我认真观察了 Ads ams api 给我的数据，通过与店铺后台广告活动（每天）、广告活动（每小时）、广告活动管理。这四份数据对比，发现只有店铺后台广告活动（每天）与广告活动管理 这两处数据勉强对得上。但是 AMS 和广告活动（每小时）与其余两个对不上，这是个什么情况呢？

**Comment by kdwivedi13** [comment]
Hi @@936447225
Thank you for your feedback. As mentioned above, our reporting API is ment for daily grain data at this time. For real-time or hourly data access, we recommend using Amazon Marketing Stream. This service provides continuous data feeds, eliminating the need to poll the API for updated data.
If you are experiencing any discrepancies with your Stream data, please create a support ticket following this [link](https://amzn-clicks.atlassian.net/servicedesk/customer/portal/2/group/2) with relevant details, and our support team will be happy to investigate. We appreciate your understanding.
